sdk/src/asset_handlers/bmff_io.rs

```diff
@@ -1390,6 +1455,119 @@ impl CAIWriter for BmffIO {
             &mut bmff_map,
         )?;
 
+        // figure out what state we are in
+        let c2pa_boxes = c2pa_boxes_from_tree_and_map(input_stream, &bmff_tree, &bmff_map)?;
+        let has_manifest = c2pa_boxes.manifest_bytes.is_some();
+        let has_original = c2pa_boxes.original_bytes.is_some();
+        let has_update = c2pa_boxes.update_bytes.is_some();
+        // if the incoming Store has an update manifest we must split it into original and update stores
+        let mut validation_log =
+            StatusTracker::with_error_behavior(ErrorBehavior::StopOnFirstError);
+        let (pc, is_update) = if let Ok(store) = Store::from_jumbf(store_bytes, &mut validation_log)
+        {
+            let pc = store
+                .provenance_claim()
+                .ok_or(Error::BadParam("no provenance claim".to_string()))?;
+            let is_update = pc.update_manifest();
+            (Some(pc.clone()), is_update)
+        } else {
+            (None, false)
+        };
+
+        // "original" manifest store and "update" manifest store can only appear together
+        if has_original && !has_update || !has_original && has_update {
+            return Err(Error::BadParam(
+                "BMFF save failure, found original manifest store without update manifest store"
+                    .to_string(),
+            ));
+        }
+
+        // if is an ordinary manifest then it should not have an update manifest
+        if has_manifest && has_update {
+            return Err(Error::BadParam(
+                "BMFF save failure, found manifest store with update manifest store".to_string(),
+            ));
+        }
+
+        // if we already have an "original" manifest store and an "update" manifest store
+        // then we can just apppend to the update store
+        if has_original && has_update && is_update {
```

Comment:
What would happen if you had a file with a split original and update manifest and wanted to append a new standard manifest on top, but there were new data boxes appended after the update manifest by the user (e.g. two videos merged)? The update manifest would no longer be the last box, does this affect the code? Is that update manifest box just deleted and appended with the new standard manifest at the end of the file?